Assets/Scripts/Boundaries.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using UnityEngine;
+
+[Serializable]
+public class Boundaries
+{
+	public Vector3 LowerBoundaries { get; private set;}
+	public Vector3 UpperBoundaries { get; private set;}
+
+	public Boundaries(Vector3 lowerBoundaries, Vector3 upperBoundaries)
+	{
+		LowerBoundaries = lowerBoundaries;
+		UpperBoundaries = upperBoundaries;
+	}
+
+
+	public Vector3 ClampToBoundaries(Vector3 value)
+	{
+		return new Vector3(
+			x: Mathf.Clamp(value.x,LowerBoundaries.x,UpperBoundaries.x),
+			y: Mathf.Clamp(value.y, LowerBoundaries.y, UpperBoundaries.y),
+			z: Mathf.Clamp(value.z, LowerBoundaries.z, UpperBoundaries.z)
+			);
```

Comment:
Maybe I just didn't see it but did you consider to substract something from the boundaries for z (if this are the map boundaries)? This is helpful to cover the whole map with the camera without allowing to move it 'into the void'.